Sources/StreamChat/APIClient/Endpoints/Payloads/IdentifiablePayload.swift

```diff
@@ -0,0 +1,224 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+protocol IdentifiablePayload {
+    var databaseId: String? { get }
+    static var modelClass: (NSManagedObject & IdentifiableModel).Type? { get }
+    func fillIds(cache: inout [String: Set<String>])
+}
+
+extension IdentifiablePayload {
+    func addId(cache: inout [String: Set<String>]) {
+        guard let databaseId = databaseId, let modelClassName = Self.modelClass?.className else { return }
+        var ids = (cache[modelClassName] ?? Set<String>())
+        ids.insert(databaseId)
+        cache[modelClassName] = ids
+    }
+
+    func recursivelyGetAllIds() -> [String: Set<String>] {
+        var cache: [String: Set<String>] = [:]
+        fillIds(cache: &cache)
+        return cache
+    }
+
+    func getPayloadToModelIdMappings(context: NSManagedObjectContext) -> PreWarmedCache {
+        let payloadIdsMappings = recursivelyGetAllIds()
+
+        var modelToIdMappings: [String: [String: NSManagedObjectID]] = [:]
+
+        for (className, identifiableValues) in payloadIdsMappings {
+            let modelClass: (NSManagedObject & IdentifiableModel).Type? = {
+                switch className {
+                case ChannelDTO.className:
+                    return ChannelDTO.self
+                case UserDTO.className:
+                    return UserDTO.self
+                case MessageDTO.className:
+                    return MessageDTO.self
+                case MessageReactionDTO.className:
+                    return MessageReactionDTO.self
+                case MemberDTO.className:
+                    return MemberDTO.self
+                case ChannelReadDTO.className:
+                    return ChannelReadDTO.self
+                default:
+                    return nil
+                }
+            }()
+
+            guard let modelClass = modelClass, let keyPath = modelClass.idKeyPath else { continue }
+
+            let values = Array(identifiableValues)
+            var results: [NSManagedObject]?
+            context.performAndWait {
+                results = modelClass.batchFetch(keyPath: keyPath, equalTo: values, context: context)
```

Comment:
You can have a `batchFetchIds` func that'll only fetch ManagedObjectIds (_how_ is explained below), will have better performance since this part doesn't require whole objects

Reply:
Let's continue the discussion in the other comment